specs/PushNotifications/ToastNotifications-spec.md

```diff
@@ -0,0 +1,509 @@
+
+The WinAppSDK AppNotification APIs enable all Desktop applications to display the same rich UI
+pop-ups that they are used to doing today minus all the complexities and overhead described above.
+Desktop applications include Unpackaged apps and Packaged win32 (MSIX Desktop Bridge, Sparse Signed
+Packages). The SDK implementation will ensure that all Notification scenarios will adhere to OS
+resource management policies like Power Saver, Network Attribution (reporting the amount of data an
+app uses), Group Policies, etc. The SDK will also abstract away the complexities of dealing with
+Notification delivery and related activations as much as possible freeing the developer to focus on
+other app related challenges.
+
+The App Notification API allows Packaged and Unpackaged apps to:
+
+-   Use common APIs to Register and Unregister Notifications.
+-   Use common APIs to display Cloud Sourced Notifications.
+-   Use common APIs to display Local Notifications.
+-   Use common APIs to Update/Delete Notifications.
+-   Use AppLifecycle constructs and event handlers to process invoke operations on behalf of the
+    user.
+
+# Examples
+
+## Registering and handling a notification
+
+Packaged and Unpackaged Desktop apps will need to register themselves to handle App Notifications.
+For Packaged apps, WinAppSDK will use the CLSID defined in the appx manifest to register the current
+process as the COM Server for Notification invokes. The process.exe declared in the manifest will
+have to be the same process calling into Register() for the API call to succeed. For Unpackaged
+apps, WinAppSDK will call into Shell libraries to determine appropriate DisplayName and and Icon.
+The current process.exe will be the COM Server for Notification invokes.
+
+```cpp
+
+int main()
+{
+    // Register an event handler before calling Register()
+    // If this is not done before calling Register(), the OS will bring up a new process everytime for a Notification Invoke.
+    const auto token = winrt::AppNotificationManager::Default().NotificationInvoked([](const auto&, const winrt::AppNotificationActivatedEventArgs& notificationActivatedEventArgs)
+    {
+        ProcessNotificationArgs(notificationActivatedEventArgs);
+    });
+
+    winrt::AppNotificationManager::Default().Register();
+
+    auto args = winrt::AppInstance::GetCurrent().GetActivatedEventArgs();
+    auto kind = args.Kind();
+
+    if (kind == winrt::ExtendedActivationKind::Launch)
+    {
+        // App does Launch Stuff Here
+    }
+    else if (kind == winrt::ExtendedActivationKind::AppNotification)
+    {
+        auto notificationActivatedEventArgs = args.Data().as<winrt::AppNotificationActivatedEventArgs>();
+        ProcessNotificationArgs(notificationActivatedEventArgs);
+    }
+
+    // Cleanup event handler
+    winrt::AppNotificationManager::Default().NotificationInvoked(token);
+
+    // Call Unregister() before exiting main so that subsequent invocations will launch a new process
```

Comment:
I imagine that during certain scenarios, the developer would want to call this if they prefer a new process launch instead of using the existing process as the COM server. So maybe not really necessary for this scenario where we bail out of main.

Reply:
If the app crashes, the COM object is destroyed for the current app and any subsequent activations will launch a new instance of the app.

If the app calls Unregister(), any activations will create a new process as the com server.